web/src/transmission.js

```diff
@@ -373,10 +373,24 @@ export class Transmission extends EventTarget {
         break;
 
       case Prefs.RefreshRate: {
-        clearInterval(this.refreshTorrentsInterval);
+        clearTimeout(this.refreshTimeoutId);
         const callback = this.refreshTorrents.bind(this);
         const msec = Math.max(2, this.prefs.refresh_rate_sec) * 1000;
-        this.refreshTorrentsInterval = setInterval(callback, msec);
+
+        const refreshTorrentsInterval = () => {
+          callback();
+          this.refreshTimeoutId = setTimeout(() => {
+            this.refreshTimeoutId = null;
+            requestAnimationFrame(() => {
+              if (this.refreshTimeoutId) {
+                return;
+              }
+              refreshTorrentsInterval();
```

Comment:
```suggestion
              if (!this.refreshTimeoutId) {
                refreshTorrentsInterval();
              }
```

Reply:
Oh, I feel silly, thank you for pointing that out. I keep forgetting to co-author you but you always suggest good changes.

Reply:
> I keep forgetting to co-author you but you always suggest good changes.

No worries. You don't need to if I didn't specifically ask for it.